packages/nx/src/utils/project-graph-utils.ts

```diff
@@ -111,8 +111,8 @@ function collectDependentProjectNodesNames(
   for (const dependency of dependencies) {
     const dependencyName = dependency.target;
 
-    // we're only intersted in project dependencies, not npm
-    if (dependencyName.startsWith('npm:')) {
+    // we're only interested in internal nodes, not external
+    if (nxDeps.externalNodes?.[dependencyName]) {
```

Comment:
We should rename nxDeps -> projectGraph or similar, I thought it was a different data structure at a glance. Either way, thats existing. Feel free to correct now or leave it, it can always be cleaned up a bit later.